File: 2020/j2.py
# code by savir singh

p = int(input())
n = int(input())
r = int(input())
day = 0
infected = n
notreached = True
infectedlist = []
infectedlist.append(infected)

if n != 1 or r != 1:
    while notreached:
        infected = infected*r
        infectedlist.append(infected)
        day += 1
        if sum(infectedlist) > p:
            notreached = False
else:
    day = p
    
print(day)

File: 2020/j2.cpp
// code by savir singh

#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main() {
    int p, n, r;
    int days=0;
    cin >> p >> n >> r;
    int originalr = r;
    int originaln = n;
    while (n<=p) {
        r=pow(originalr, days+1);
        r*=originaln;
        n+=r;
        days++;
    }
    cout << days;
}
